src/components/RichTextEditor/RichTextEditor.tsx

```diff
@@ -1,156 +1,132 @@
 /* (c) Copyright Frontify Ltd., all rights reserved. */
 
-import { getMinWidthIfEmpty } from "@components/RichTextEditor/utils/getMinWidthIfEmpty";
-import { compose } from "@utilities/compose";
+import {
+    BalloonToolbar,
+    createAlignPlugin,
+    createBoldPlugin,
+    createCodeBlockPlugin,
+    createCodePlugin,
+    createHeadingPlugin,
+    createIndentListPlugin,
+    createIndentPlugin,
+    createItalicPlugin,
+    createLinkPlugin,
+    createParagraphPlugin,
+    createPlateUI,
+    createPlugins,
+    createSoftBreakPlugin,
+    createStrikethroughPlugin,
+    createUnderlinePlugin,
+    ELEMENT_H1,
+    ELEMENT_LINK,
+    ELEMENT_PARAGRAPH,
+    MARK_BOLD,
+    MARK_ITALIC,
+    MARK_STRIKETHROUGH,
+    MARK_UNDERLINE,
+    Plate,
+    usePlateEditorRef,
+} from "@udecode/plate";
 import { debounce } from "@utilities/debounce";
-import { useDebounce } from "@utilities/useDebounce";
-import { useMachine } from "@xstate/react";
-import React, { CSSProperties, FC, useCallback, useEffect, useMemo, useRef, useState } from "react";
-import { BaseEditor, createEditor, Descendant, Transforms } from "slate";
-import { withHistory } from "slate-history";
-import { Editable, ReactEditor, Slate, withReact } from "slate-react";
-import { DoneInvokeEvent, Interpreter } from "xstate";
-import { ToolbarContext } from "./context/toolbar";
-import { useSoftBreak } from "./hooks/useSoftBreak";
-import { withLinks } from "./plugins/withLinks";
-import { withLists } from "./plugins/withLists";
-import { BlockStyleTypes, renderBlockStyles, TextAlignTypes } from "./renderer/renderBlockStyles";
-import { InlineStyles, renderInlineStyles } from "./renderer/renderInlineStyles";
-import { editorMachine, States } from "./state/editor/machine";
-import { ToolbarContext as ToolbarFSMContext, ToolbarData } from "./state/toolbar/machine";
+import React, { FC, useEffect } from "react";
+import { EditableProps } from "slate-react/dist/components/editable";
+import { LinkElement } from "./components/elements/link";
+import { BoldMark } from "./components/marks/bold";
+import { ItalicMark } from "./components/marks/italic";
+import { StrikethroughMark } from "./components/marks/strikethrough";
+import { UnderlineMark } from "./components/marks/underline";
 import { Toolbar } from "./Toolbar";
-import { clearEditor } from "./utils/editor/clear";
-import { parseRawValue } from "./utils/editor/parseRawContent";
+import { clearEditor } from "./utils/clearEditor";
 
 export type RichTextEditorProps = {
     placeholder?: string;
-    value?: string;
+    value?: any;
```

Comment:
the idea there was to provide a serialized json object as string, as we provide also a serialized json object in the onChange-/onBlur-event

Reply:
Changed it back to serialized json